code/game/machinery/planet_beacon.dm

```diff
@@ -0,0 +1,216 @@
+					return
+				if(change_state(BEACON_DISTRESS))
+					play_click_sound()
+					to_chat(user, "<span class='danger'>You activate the distress beacon!</span>")
+
+			if("Turn off beacon")
+				to_chat(user, "<span class='notice'>You turn off the beacon.</span>")
+				playsound(src, 'sound/machines/terminal_off.ogg', 50, FALSE)
+				change_state(BEACON_ANCHORED)
+
+/obj/machinery/power/planet_beacon/AltClick(mob/user)
+	..()
+	if(state != BEACON_UNANCHORED)
+		if(inserted_id)
+			try_put_in_hand(inserted_id, user)
+			inserted_id = null
+			update_icon()
+			playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+			return
+
+/obj/machinery/power/planet_beacon/examine(mob/user)
+	. = ..()
+	if(inserted_id)
+		. += "<span class='notice'>Alt-click to eject the ID card.</span>"
+	if(preserving)
+		. += "<span class='notice'>It is currently active, marking the zone for future docking.</span>"
+	switch(state)
+		if(BEACON_UNANCHORED)
+			. += "<span class='notice'>It is currently unsecured, and must be wrenched in place to use.</span>"
+		if(BEACON_ANCHORED)
+			. += "<span class='notice'>It is currently secured, and can be activated.</span>"
+		// Active message handled by preserving
+		if(BEACON_DISTRESS)
+			. += "<span class='warning'>It is currently broadcasting a distress signal!</span>"
+
+/obj/machinery/power/planet_beacon/proc/change_state(new_state)
+	if(!new_state)
+		return
+
+	switch (new_state)
+		if(BEACON_UNANCHORED) // Unanchored, cannot be turned on
+			icon_state = "beacon"
+			preserving = FALSE
+			set_light(0)
+		if(BEACON_ANCHORED) // Anchored, able to be turned on
+			icon_state = "beacon_anchored"
+			preserving = FALSE
+			set_light(0)
+		if(BEACON_IDLE) // Turning on, but not persistent yet, idle power usage
+			icon_state = "beacon_idle"
+			preserving = FALSE
+			set_light_color(COLOR_CYAN)
+			set_light(2)
+		if(BEACON_ACTIVE) // Enable persistence, use full power
+			icon_state = "beacon_active"
+			preserving = TRUE
+			set_light_color(COLOR_CYAN)
+			set_light(2)
+		if(BEACON_DISTRESS) // Send out a distress and temporarily change the screen
+			if(!send_distress())
+				return FALSE
+			icon_state = "beacon_distress"
+			set_light_color(COLOR_DARK_RED)
+			set_light(3)
+
+			addtimer(CALLBACK(src, PROC_REF(change_state), state), 10 SECONDS) // Return to the previous state after a certain time
+
+	state = new_state // Actually change the state, can you believe I forgot this at one point?
+	referesh_preservation()
+	return TRUE
+
+/obj/machinery/power/planet_beacon/update_overlays()
+	. = ..()
+	if(inserted_id)
+		. += mutable_appearance(icon, "overlay_card")
+
+/obj/machinery/power/planet_beacon/proc/send_distress()
+	if(!inserted_id)
+		return FALSE
+
+	var/datum/overmap/dynamic/OE = get_overmap_location()
+
+	var/text = "A distress beacon has been activated by [inserted_id.registered_name] at local sector co-ordinates [OE.x]/[OE.y]. No further information available."
```

Comment:
If possible, please convert this to use `proc/create_distress_beacon`. The original proc may need some editing but it should be used here.